### PR TITLE
#7178: falcon7b - fix expected pcc for test_perf_falcon

### PR DESCRIPTION
**Pull request by uaydonat**
Enhancing pcc checks for test_perf_falcon variants.

1. Separate expected pcc's for output, k cache, and v cache.
2. pcc checks fail if device pcc is not within the 2nd decimal
3. higher or lower pcc will fail; expected pcc should be updated if pcc is higher than expected.

### CODE REVIEW

**Review by skhorasganiTT — APPROVED** [review]

File: models/demos/falcon7b/tests/test_perf_falcon.py [hunk 1]
```diff
@@ -331,31 +358,26 @@ def run_test_FalconCausalLM_end_to_end(
...
-            ("prefill", 1, 128, 0, 0.30),
-            ("prefill", 1, 256, 0, 0.44),
-            ("decode", 32, 1, 128, 0.27),
-            ("decode", 32, 1, 1024, 0.35),
-            ("decode", 32, 1, 2047, 0.48),
+            ("prefill", 32, 1, 128, 0, "BFLOAT16-L1", 0.85, 0.97, 0.86, 0.30),
+            ("prefill", 32, 1, 256, 0, "BFLOAT16-L1", 0.90, 0.97, 0.87, 0.44),
+            ("decode", 32, 32, 1, 128, "BFLOAT16-L1", 0.59, 0.73, 0.79, 0.27),
+            ("decode", 32, 32, 1, 1024, "BFLOAT16-L1", 0.59, 0.85, 0.85, 0.35),
+            ("decode", 32, 32, 1, 2047, "BFLOAT16-L1", 0.55, 0.94, 0.91, 0.48),
```
[the remark this answers is not in the document]
Comment (uaydonat) [reply models/demos/falcon7b/tests/test_perf_falcon.py:380]: yes, I will add dram back in addition to L1 here.

File: models/demos/falcon7b/tests/test_perf_falcon.py [hunk 2]
```diff
@@ -464,20 +492,38 @@ def run_perf_wh_bare_metal(
...
+            "prefill_seq256_bf16_l1",
+            "prefill_seq256_bf16_l1_sharded",
+            "decode_batch32_128_bf16_dram",
+            "decode_batch32_128_bf16_l1",
+            "decode_batch32_128_bf16_l1_sharded",
+            "decode_batch32_1024_bf16_dram",
+            "decode_batch32_1024_bf16_l1",
+            "decode_batch32_1024_bf16_l1_sharded",
+            "decode_batch32_2047_bf16_dram",
+            "decode_batch32_2047_bf16_l1",
```
[the remark this answers is not in the document]
Comment (uaydonat) [reply models/demos/falcon7b/tests/test_perf_falcon.py:526]: Good catch. I actually added those as sanity, L1_sharded is equal to L1 for prefill. I will remove them.

**Comment by uaydonat** [comment]
Pipelines (both single chip and multi-chip) passed for falcon7:

https://github.com/tenstorrent/tt-metal/actions/runs/8902088945
https://github.com/tenstorrent/tt-metal/actions/runs/8902092079

File: models/demos/falcon7b/tests/test_perf_falcon.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (uaydonat) [reply models/demos/falcon7b/tests/test_perf_falcon.py]: done

File: models/demos/falcon7b/tests/test_perf_falcon.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (uaydonat) [reply models/demos/falcon7b/tests/test_perf_falcon.py]: Added dram in addition to L1

**Comment by uaydonat** [comment]
can I get a review?

@eyonland @cfjchu @arakhmati @xanderchin

**Review by arakhmati — APPROVED** [review]

**Pull request merged**